src/snowflake/snowpark/mock/_stage_registry.py

```diff
@@ -413,6 +444,80 @@ def read_file(
                 result_df = pd.concat([result_df, df], ignore_index=True)
             result_df.sf_types = result_df_sf_types
             return result_df
+        elif file_format == "json":
+            infer_schema_opt = options.get("INFER_SCHEMA", False)
+
+            result_df = TableEmulator()
+            result_df_sf_types = {}
+
+            if not infer_schema_opt:
+                for i in range(len(schema)):
+                    column_name = analyzer.analyze(schema[i])
+                    column_series = ColumnEmulator(
+                        data=None, dtype=object, name=column_name
+                    )
+                    column_series.sf_type = ColumnType(
+                        schema[i].datatype, schema[i].nullable
+                    )
+                    result_df[column_name] = column_series
+                    result_df_sf_types[column_name] = column_series.sf_type
+
+                for local_file in local_files:
+                    with open(local_file) as file:
+                        content = json.load(file)
+                        df = pd.DataFrame({result_df.columns[0]: [content]})
+                        result_df = pd.concat([result_df, df], ignore_index=True)
+            else:
+                # need to infer schema
+                contents = []
+                for local_file in local_files:
+                    with open(local_file) as file:
+                        content = json.load(file)
+                        contents.append(content)
+                        # if not result_df_sf_types:
+                        # extract the schema from the content
+                        schema = infer_schema(content)
+                        for field in schema:
+                            lower_filed_name = field.name.lower()
+                            column_name = (
+                                lower_filed_name
+                                if lower_filed_name in content
+                                else file.name
```

Comment:
Why substitute in the file name here?

Reply:
thanks for catching this part, I realized that there is a bug here that the implementation can not handle mixed-case column name.
I reimplemented this part and updated the test file to test the case.